commit b45bc4e38f404966d7f538ed7add6ea647f36903
Author: Yang Bo <yangbo@deepin.com>
Date:   Fri Jun 9 09:56:09 2017 +0800

    ipv6: Fix leak in ipv6_gso_segment().
    
    If ip6_find_1stfragopt() fails and we return an error we have to free
    up 'segs' because nobody else is going to.
    
    Signed-off-by: Yang Bo <yangbo@deepin.com>

diff --git a/net/ipv6/ip6_offload.c b/net/ipv6/ip6_offload.c
index 9881a8769..20ad65526 100644
--- a/net/ipv6/ip6_offload.c
+++ b/net/ipv6/ip6_offload.c
@@ -117,8 +117,10 @@ static struct sk_buff *ipv6_gso_segment(struct sk_buff *skb,
 
 		if (udpfrag) {
 			unfrag_ip6hlen = ip6_find_1stfragopt(skb, &prevhdr);
-			if (unfrag_ip6hlen < 0)
+			if (unfrag_ip6hlen < 0) {
+				kfree_skb_list(segs);
 				return ERR_PTR(unfrag_ip6hlen);
+			}
 			fptr = (struct frag_hdr *)((u8 *)ipv6h + unfrag_ip6hlen);
 			fptr->frag_off = htons(offset);
 			if (skb->next)
